dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/data/NeuralNetworkDataParser.java

```diff
@@ -83,13 +90,33 @@ public void parse() {
         final Matrix data = input.get(range(0, input.getColumns() - 2));
         final int label = (int) input.get(input.getColumns() - 2);
         final boolean isValidation = ((int) input.get(input.getColumns() - 1) == 1);
-        trainingData.add(new Pair<>(new Pair<>(data, label), isValidation));
+
+        if (isValidation) {
+          final Pair<Matrix, int[]> batch = validationBatchGenerator.push(data, label);
```

Comment:
Making `push` return a value seems a bit awkward. Does `BatchGenerator` have to be separated from this `NeuralNetworkDataParser` class, like the current code? Wouldn't be it okay to make `BatchGenerator` directly add pairs to `dataList` when the batch size is reached? The `matrixFactory` field of  `NeuralNetworkDataParser` could directly be used, too.